adapters/handlers/grpc/server.go

```diff
@@ -29,10 +30,24 @@ import (
 const maxMsgSize = 104858000 // 10mb, needs to be synchronized with clients
 
 func CreateGRPCServer(state *state.State) *GRPCServer {
-	s := grpc.NewServer(
+	o := []grpc.ServerOption{
 		grpc.MaxRecvMsgSize(maxMsgSize),
 		grpc.MaxSendMsgSize(maxMsgSize),
-	)
+	}
+
+	// Add TLS creds for the GRPC connection, if defined.
+	if len(state.ServerConfig.Config.GRPC.CertFile) > 0 && len(state.ServerConfig.Config.GRPC.KeyFile) > 0 {
+		c, err := credentials.NewServerTLSFromFile(state.ServerConfig.Config.GRPC.CertFile,
+			state.ServerConfig.Config.GRPC.KeyFile)
+		if err != nil {
+			state.Logger.WithField("action", "grpc_startup").
+				Errorf("grpc server TLS credential error: %s", err)
```

Comment:
This should be changed to a `Fatalf` error. Otherwise, the server will not initialize, and startup will result in a nil pointer panic:

```
ERRO[0001] grpc server TLS credential error: open /Users/parkerduckworth/.ssh/localhost/localhoasdft.cer.pem: no such file or directory  action=grpc_startup
INFO[0001] grpc server listening at [::]:50051           action=grpc_startup
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x103b6dff8]

goroutine 972 [running]:
github.com/weaviate/weaviate/adapters/handlers/grpc.StartAndListen(0x0, 0x1400359bba0)
        /Users/parkerduckworth/semi/weaviate/adapters/handlers/grpc/server.go:78 +0xf8
github.com/weaviate/weaviate/adapters/handlers/rest.startGrpcServer.func1()
        /Users/parkerduckworth/semi/weaviate/adapters/handlers/rest/grpc.go:25 +0x30
created by github.com/weaviate/weaviate/adapters/handlers/rest.startGrpcServer in goroutine 1
        /Users/parkerduckworth/semi/weaviate/adapters/handlers/rest/grpc.go:24 +0x70
exit status 2
```

If a cert/key were provided via envvar, the intention of the cluster admin was to run a secure server. IMO it's better to fail immediately, than to log an error message which may be missed, and continue serving an unsecured API

Reply:
Thanks, makes sense